### ISSUE TITLE
[Question] Maturity?

### ISSUE THREAD

**Issue by skyzyx** [digiaonline/graphql-php#28]
My team has been using Youshido GraphQL for a project over the last 9 months or so. It gets the job done, but it's slow, not documented well, and PRs take _forever_ to get merged and released. So my team has been debating the pros/cons of forking it and driving it in a more performant direction.

One of the performance-related PRs on that project appears to have come from the developers of this project… which leads me to ask some questions:

1. Is this project ready for heavy production loads by teams outside of yourselves?

1. Is there a migration path from Youshido to this one?

1. What kind of stability/support would another team be able to expect from this package?

1. Are you open to third-party PRs, and is there a reasonably straightforward process for contributing back upstream?

In other words, how _mature_ is this as an open-source project?

**Comment by Jalle19** [comment]
Hi @skyzyx,

Thanks for noticing my PR! It is indeed a shame that Youshido is pretty slow at merging PRs, that is one of the reasons why we decided to write our own library. I'll try to answer some of your questions:

1. We're still not using it anywhere since it's still under development, but the idea is to introduce it in some of our smaller projects first and then (hopefully) in larger projects too. We've been using Youshido so far.

2. @crisu83 can perhaps answer that, but my gut feeling is "no". It's still GraphQL though so a lot of the concepts are the same, so porting over an existing application shouldn't be an insurmountable task.

3. Generally when we make open source packages it's because we use them in one or more projects, so in that sense you can expect stability and support.

4. Definitely, although I think it's still fairly early since this project is still under heavy development. @crisu83 can maybe clarify this a bit.

Thanks for showing interest in our library!

**Comment by crisu83** [comment]
Hello @skyzyx,

I'll try my best to fill in the rest of the answers.

1. Yes, but there it's not finished yet. We're working really hard on getting the first version ready.
2. Our library uses Schema Definition Langauge (SDL) to define the schema, but migrating shouldn't be hard. Basically, you'd need to migrate your definitions to SDL and move your resolving logic into resolvers (this part has not yet been implemented).
3. We are building this library for production use in many upcoming projects so you can expect stable (semver) releases. We also hope that many communities will adopt our implementation (Laravel, Symfony, etc.).
4. Yes. However, the contributed code needs to follow the guidelines we set for the project (we'll add this shortly).

Thanks on my behalf as well. I'll be happy to invite you into our Slack if you drop me a line at christofferniska@gmail.com and we can continue the discussion there.

**Comment by fubhy** [comment]
Hi. Maintainer of https://github.com/drupal-graphql/graphql here. I would love to get an invite to your slack too: fubhy@fubhy.com

**Comment by crisu83** [comment]
@fubhy Sent you an invite.

**Comment by crisu83** [comment]
@skyzyx are you still interested in this project? We've come quite far since we last exchanged messages.

**Comment by skyzyx** [comment]
@crisu83: Definitely. We started out by forking the Youshido library and making a variety of optimizations. If we were to migrate, we'd need to spend the time looking at a stable release to determine how to port our existing API.

Our GraphQL service drives our UI and some developer tools at our company. Some things are easy because we have our own database backend, but parts of the API proxy back to another backend API that is Restful. Because of this, we make heavy use of deferred resolvers with lots of caching. We have a lot of hand-written code to manage all of this, so simplifying that would be nice.

Performance is also a huge concern. The Youshido library is unnecessarily slow in certain areas, and we've had to do a lot of optimizations to improve the performance. We also pulled-in the PR that was provided by you guys, which resulted in a dramatic improvement (thank you!). It doesn't help that their library is virtually unmaintained.

But really, it's about having a solid migration path from Youshido to this library. Preferably something simple enough to justify to management that it's low-risk, high-reward.

**Comment by fubhy** [comment]
Migrating to this library should be fairly simple since you'd just be defining your schema through the SDL and then moving your existing resolver classes / functions into the resolver map. We should maybe consider writing some migration docs?

**Comment by crisu83** [comment]
Closing due to inactivity.
